app/controllers/form_controller.rb

```diff
@@ -89,7 +89,7 @@ def responses_for_page(page)
         year = params["case_log"]["#{question.id}(1i)"]
         next unless [day, month, year].any?(&:present?)
 
-        result[question.id] = if day.to_i.between?(1, 31) && month.to_i.between?(1, 12) && year.to_i.between?(2000, 2200)
+        result[question.id] = if day.to_i.between?(1, 31) && month.to_i.between?(1, 12) && year.to_i.between?(2000, 2200) && Time.days_in_month(month.to_i, year.to_i) >= day.to_i
```

Comment:
```suggestion
        result[question.id] = if Date.valid_date?(year.to_i, month.to_i, day.to_i) && year.to_i.between?(2000, 2200)
```

Do you think this is a little easier to parse?